src/d3a/d3a_core/redis_communication.py

```diff
@@ -101,17 +103,36 @@ def _handle_redis_job_metadata(self):
             self._simulation.stop()
 
     def publish_results(self, endpoint_buffer):
-        if not hasattr(self, 'pubsub'):
+        if not self.is_enabled():
             return
         results = endpoint_buffer.generate_result_report()
         results_validator(results)
         self.redis_db.publish(self.result_channel, json.dumps(results))
         self._handle_redis_job_metadata()
 
+    def write_zip_results(self, zip_results):
+        if not self.is_enabled():
+            return
+
+        fp = open(zip_results, 'rb')
+        zip_data = fp.read()
+        fp.close()
+
+        zip_results_key = ZIP_RESULTS_KEY + str(self._simulation_id)
+        # Write results to a separate Redis key
+        self.redis_db.set(zip_results_key, zip_data)
```

Comment:
I guess it could be available to all channels. Right?

Reply:
Sorry, same as the other PR, this is not a channel but a Redis key :)

Reply:
Thanks for an offline explanation 👍